### PR TITLE
Mobile memory disposal pattern+UI elements

### PR DESCRIPTION
**Pull request by Felicity-R**
Document memory management and more UI elements for mobile:
https://geocortex.visualstudio.com/Geocortex/_workitems/edit/40023
https://geocortex.visualstudio.com/Geocortex/_workitems/edit/41574/

### CODE REVIEW

**Review by rcooney — APPROVED** [review]

**Review by jsiemensLatitudegeo — CHANGES_REQUESTED** [review]

File: docs/mobile/sdk-memory-leaks.mdx [hunk 1]
```diff
@@ -3,8 +3,41 @@ title: Memory Management
...
+```cs
+public class DisposableTrackerExample: IDisposableTracker, IDisposable
+{
+	// Implement the IDisposableTracker interface
+    public IList<IDisposable> Disposables => new List<IDisposable>();
+
+	public DisposableTrackerExample()
+	{
+		// Pointer to a fictitious external unmanaged resource.
+        SafeFileHandle handle = new SafeFileHandle(IntPtr.Zero, true);
```
[the remark this answers is not in the document]
Comment (Felicity-R) [reply docs/mobile/sdk-memory-leaks.mdx]: Fixed

File: docs/mobile/sdk-components-styling.mdx [hunk 2]
```diff
@@ -13,6 +13,87 @@ Geocortex Mobile Components are styled using [XAML styles](https://docs.microsof
 
 Geocortex Mobile XAML Elements, like the `EnhancedButton` or `EnhancedSwitch` can ease the development of custom components with the SDK. Check out the [relevant SDK sample](https://github.com/geocortex/vertigis-mobile-samples/tree/master/Geocortex.Mobile.Samples/Geocortex.Mobile.Samples/Samples/Custom/GeocortexMobileElements) for an end to end example of their use.
```
[the remark this answers is not in the document]
Comment (Felicity-R) [reply docs/mobile/sdk-components-styling.mdx]: Changed 'XAML Elements' to 'Views' throughout this page

File: docs/mobile/sdk-memory-leaks.mdx [hunk 3]
```diff
@@ -3,8 +3,41 @@ title: Memory Management
...
+It is important to ensure that unmanaged resources are released when they are no longer required, in order to avoid creating memory leaks. This is accomplished using the `IDisposable` interface.
+If a class creates unmanaged resources, or creates instances of classes which implement `IDisposable` themselves, it should implement `IDisposable` and release these resources in the `Dispose()` method. Detailed information on how to implement `Dispose()` can be found in [Microsoft's documentation, here](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose).
 
-This documentation is under active development.
+:::note
+Classes instantiated in Mobile via [dependency injection](sdk-dependency-injection.mdx) will automatically be disposed by Autofac, and do not need to be disposed of manually by the developer.
+:::
 
-For now, check out the sample on the [disposal pattern in Geocortex Mobile](https://github.com/geocortex/vertigis-mobile-samples/tree/master/Geocortex.Mobile.Samples/Geocortex.Mobile.Samples/Samples/Conceptual/Disposal).
+## Using Mobile's IDisposableTracker Interface
```
[the remark this answers is not in the document]
Comment (Felicity-R) [reply docs/mobile/sdk-memory-leaks.mdx:13]: Added brief note with links to the Commands and Operations and Events sections

File: docs/mobile/sdk-memory-leaks.mdx [hunk 4]
```diff
@@ -3,8 +3,41 @@ title: Memory Management
 description: Geocortex Mobile - Learn about memory management in Geocortex Mobile
 ---
 
-# Coming Soon...
+It is important to ensure that unmanaged resources are released when they are no longer required, in order to avoid creating memory leaks. This is accomplished using the `IDisposable` interface.
+If a class creates unmanaged resources, or creates instances of classes which implement `IDisposable` themselves, it should implement `IDisposable` and release these resources in the `Dispose()` method. Detailed information on how to implement `Dispose()` can be found in [Microsoft's documentation, here](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose).
 
-This documentation is under active development.
+:::note
+Classes instantiated in Mobile via [dependency injection](sdk-dependency-injection.mdx) will automatically be disposed by Autofac, and do not need to be disposed of manually by the developer.
```
[the remark this answers is not in the document]
Comment (Felicity-R) [reply docs/mobile/sdk-memory-leaks.mdx]: Reworded in a way which should make it clearer that it's just the call to Dispose() that's taken care of by AutoFac, not the actual implementation of IDisposable.

File: docs/mobile/sdk-memory-leaks.mdx [hunk 5]
```diff
@@ -3,8 +3,41 @@ title: Memory Management
...
+:::
 
-For now, check out the sample on the [disposal pattern in Geocortex Mobile](https://github.com/geocortex/vertigis-mobile-samples/tree/master/Geocortex.Mobile.Samples/Geocortex.Mobile.Samples/Samples/Conceptual/Disposal).
+## Using Mobile's IDisposableTracker Interface
+
+Mobile's toolkit also includes a helpful interface and extension methods to make tracking disposable resources easier: `IDisposableTracker` and `DisposableUtilities`.
+`ComponentBase` and `ServiceBase` already implement `IDisposableTracker`, so any custom components or services extending these base classes can take advantage of this functionality right away. Here is a brief example of how to use it in a new class:
```
[the remark this answers is not in the document]
Comment (Felicity-R) [reply docs/mobile/sdk-memory-leaks.mdx]: Added a note to this effect

**Review by jsiemensLatitudegeo — APPROVED** [review]

**Pull request merged**